docs/dev/relay_pass_infra.rst

```diff
@@ -0,0 +1,616 @@
+..  Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+..    http://www.apache.org/licenses/LICENSE-2.0
+
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Relay Pass Infra
+==================================
+
+Relay features a series of optimization passes which improve performance metrics
+of models such as mean inference, memory footprint, or power consumption for
+specific devices. There are a suite of standard optimizations as well as machine
+learning specific optimizations including constant folding, dead code
+elimination, operator layout alteration, and operator fusion, etc. Each of these
+passes are structured as Relay-to-Relay transformations on the abstract syntax
+tree (AST) using the analysis result collected during and/or before traversal.
+
+On one hand, many existing production compilers, such as GCC and LLVM, employ
+pass managers to effectively manage the execution of passes. Initially managing
+passes is straightforward as the number of passes is small, but mature compilers
+will contain hundreds of individual passes. Often external users will want to
+have custom passes correctly scheduled without having to modify a single
+handcrafted pass order.
+
+On the other hand, modern deep learning frameworks, such as Pytorch and MXNet
+Gluon, also have the tendency to enable pass style layer construction
+scheme through `Sequential`_ and `Block`_, respectively. With such constructs,
+these modern frameworks are able to conveniently add modules/layers to their
+containers and build up neural network easily.
+
+The design of the Relay pass infra is largely inspired by the the hierarchical
+pass manager used in LLVM and the block style containers used in the popular
+deep learning frameworks. The major goals of the pass infra include:
```

Comment:
Are the two paragraphs before this one meant to set up a contrast between the two approaches? If so, this should explain how Relay's pass manager borrows from both.

Reply:
Thank you for pointing out. I am actually not trying to setup a contrast. Instead, I want to say that we considered the advantages of both other approaches. I will rephrase.